src/v2/view-builder/views/password/EnrollAuthenticatorPasswordView.js

```diff
@@ -73,6 +73,18 @@ const Body = BaseForm.extend({
         }
       }
     ]);
+  },
+
+  saveForm() {
+    let processCredsPromise = Promise.resolve();
+    const username = this.options.appState.get('user')?.profile?.login;
+    if (username) {
+      const password = this.model.get('credentials.passcode');
+      processCredsPromise = this.settings.processCreds({ username, password });
+    }
+    processCredsPromise.then(() => {
+      BaseForm.prototype.saveForm.apply(this, arguments);
+    });
```

Comment:
It looks like the Settings method `processCreds` takes care of ensuring `this.settings.processCreds` always returns a promise.

Reply:
And same thing about the identifer check here (let processCreds handle it either on our end of theirs)